### PR TITLE
chore(mc): Fix sass watch task not to create unexpected files on errors

### PR DESCRIPTION
**Pull request by k88hudson**
This fix replaces the built-in sass watch task (which targets each sass file individually for some reason) with nodemon. It prevents extra files being created and doesn't throw those weird errors every time you change an included file.

### CODE REVIEW

**Comment by Mardak** [comment]
What's the expected output with the change? Here's my before and after of changing a .scss to have an invalid variable:

Before (not actually undefined "undefined variable"):
```
=> changed: system-addon/content-src/components/TopSites/TopSites.scss
=> changed: system-addon/content-src/activity-stream.scss
{
  "status": 1,
  "file": "system-addon/content-src/components/TopSites/TopSites.scss",
  "line": 2,
  "column": 20,
  "message": "Undefined variable: \"$grid-unit\".",
  "formatted": "Error: Undefined variable: \"$grid-unit\".\n        on line 2 of system-addon/content-src/components/TopSites/TopSites.scss\n>>   $top-sites-size: $grid-unit;\n   -------------------^\n"
}
{
  "status": 1,
  "file": "system-addon/content-src/components/TopSites/TopSites.scss",
  "line": 10,
  "column": 23,
  "message": "Undefined variable: \"$ase-gutter\".",
  "formatted": "Error: Undefined variable: \"$ase-gutter\".\n        on line 10 of system-addon/content-src/components/TopSites/TopSites.scss\n>>   margin-inline-end: -$ase-gutter;\n   ----------------------^\n"
}
```

After (npm spew):
```
{
  "status": 1,
  "file": "system-addon/content-src/components/TopSites/TopSites.scss",
  "line": 10,
  "column": 23,
  "message": "Undefined variable: \"$ase-gutter\".",
  "formatted": "Error: Undefined variable: \"$ase-gutter\".\n        on line 10 of system-addon/content-src/components/TopSites/TopSites.scss\n>>   margin-inline-end: -$ase-gutter;\n   ----------------------^\n"
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! activity-streams@1.10.1 buildmc:css: `node-sass system-addon/content-src/activity-stream.scss -o system-addon/data/content`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the activity-streams@1.10.1 buildmc:css script 'node-sass system-addon/content-src/activity-stream.scss -o system-addon/data/content'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the activity-streams package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-sass system-addon/content-src/activity-stream.scss -o system-addon/data/content
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs activity-streams
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls activity-streams
npm ERR! There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     .npm/_logs/2017-05-15T18_39_48_040Z-debug.log
```

File: yamscripts.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ scripts:
     pre: =>prebuildmc
     copy: =>buildmc:copy -- -w
     webpack: =>buildmc:webpack -- -w
-    css: =>buildmc:css && =>buildmc:css -- -w
+    css: =>buildmc:css && nodemon --ext scss -x "npm run buildmc:css" --watch system-addon/content-src -q
```
Comment (Mardak) [inline yamscripts.yml]: Looks like we can silent the rest of npm error with a `-s`, so `-x "npm run buildmc:css -s"`

**Comment by Mardak** [comment]
@k88hudson found some docs saying just need to prepend with `_` to avoid random files: http://sass-lang.com/guide#topic-4

**Comment by as-pine-proxy** [comment]
[Treeherder: c305d38a9c2b2fb072d587291b3e18519136c55f](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=c305d38a9c2b2fb072d587291b3e18519136c55f)

**Pull request merged**